torchtune/modules/loss/cross_entropy_loss.py

```diff
@@ -238,6 +229,15 @@ def forward(
         if isinstance(hidden_states, DTensor):
             hidden_states = hidden_states.full_tensor()
 
+        batch_size, seq_len, emb_dim = hidden_states.shape
+        hidden_states = hidden_states.reshape(
+            -1, emb_dim
+        )  # [batch_size*seq_len, emb_dim]
+        targets = targets.reshape(-1)  # [batch_size*seq_len]
```

Comment:
Feels like since you don't reuse B, T, C values anywhere, it could be done simpler:

```suggestion
        hidden_states = hidden_states.flatten(0, 1) # (batch_size*seq_len, hidden_size)
        targets = targets.flatten() # (batch_size*seq_len)
```